.github/workflows/deploy_devnet.yml

```
@@ -0,0 +1,230 @@
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+        
+permissions:
+  id-token: write
+  contents: write
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      # Configure AWS credentials based on network.
+      - name: Checkout Code Repository
+        uses: actions/checkout@v3
+
+      - name: Configure devnet AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: eu-west-1
+          role-to-assume: arn:aws:iam::657556092833:role/midendev-GithubActionsRole
+          role-session-name: GithubActionsSession
+
+      - name: Stop services
+        id: stop_services
+        run: |
+          COMMAND_ID=$(aws ssm send-command \
+            --instance-ids ${{ secrets.DEVNET_INSTANCE_ID }} \
+            --document-name "AWS-RunShellScript" \
+            --parameters '{"commands":["sudo systemctl stop miden-node", "sudo systemctl stop miden-faucet"]}' \
+            --output text \
+            --query "Command.CommandId")
+          echo "command_id=$COMMAND_ID" >> $GITHUB_OUTPUT
+
+      - name: Check stop services command status and retrieve output
+        run: |
+          sleep 5
+          STATUS=$(aws ssm list-command-invocations \
+              --command-id ${{ steps.stop_services.outputs.command_id }} \
+              --details \
+              --query "CommandInvocations[0].Status" \
+              --output text)
+          echo "Command Status: $STATUS"
+          OUTPUT=$(aws ssm list-command-invocations \
+              --command-id ${{ steps.stop_services.outputs.command_id }} \
+              --details \
+              --query "CommandInvocations[0].CommandPlugins[0].Output" \
+              --output text)
+          echo "Command Output: $OUTPUT"
+          if [ "$STATUS" != "Success" ]; then
+              echo "Command failed with status: $STATUS"
+              exit 1
+          fi
+      
+      - name: Install prerequisites
+        id: install_prereq
+        run: |
+          COMMAND_ID=$(aws ssm send-command \
+            --instance-ids ${{ secrets.DEVNET_INSTANCE_ID }} \
+            --document-name "AWS-RunShellScript" \
+            --parameters '{"commands":[
+                  "sudo apt-get update -y",
+                  "sudo apt-get install -y git gcc openssl libbz2-dev libffi7 make cargo",
+                  "sudo apt install -y build-essential",
+                  "curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y",
+                  "source $HOME/.cargo/env",
+                  "rustup update",
+                  "sudo apt-get install -y clang cmake",
+                  "sudo rm -rf /opt/miden/miden-node/genesis.dat",
+                  "sudo rm -rf /opt/miden/miden-node/miden-store.sqlite3",
+                  "sudo rm -rf /opt/miden/miden-node/miden-store.sqlite3-*",
+                  "sudo rm -rf /opt/miden/miden-node/accounts",
+                  "sudo rm -rf /opt/miden/miden-node/blocks"
```

```suggestion
                  "sudo rm -rf /opt/miden/miden-node/",
                  "sudo mkdir  /opt/miden/miden-node/",
```
Probably best to nuke the whole directory in case we change the data required.